### PR TITLE
[KYUUBI #2500] Command OptionParser for launching Flink Backend Engine

### LINKED ISSUE #2500: [Improvement] Command OptionParser for launching Flink Backend Engine
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?
after 
https://github.com/apache/incubator-kyuubi/pull/2418
finished
we should change Flink Engine as  hive does in
https://github.com/apache/incubator-kyuubi/pull/2356 

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2500 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by jiaoqingbo** [comment]
cc @yaooqinn

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala [hunk 1]
```diff
@@ -54,27 +54,22 @@ class FlinkProcessBuilder(
     val buffer = new ArrayBuffer[String]()
     buffer += s"bash"
     buffer += s"-c"
-    val commandStr = new StringBuilder()
+    val commandSet = new LinkedHashSet[String]()
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:57]: why not use the `buffer`

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala [hunk 2]
```diff
@@ -54,27 +54,22 @@ class FlinkProcessBuilder(
...
-    commandStr.append(s"source $FLINK_HOME${File.separator}bin" +
+    commandSet.add(s"source $FLINK_HOME${File.separator}bin" +
       s"${File.separator}config.sh && $executable")
 
     // TODO: How shall we deal with proxyUser,
     // user.name
     // kyuubi.session.user
     // or just leave it, because we can handle it at operation layer
-    commandStr.append(s" -D$KYUUBI_SESSION_USER_KEY=$proxyUser ")
+    commandSet.add(s"-D$KYUUBI_SESSION_USER_KEY=$proxyUser")
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:66]: why not pass via --conf

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala (hunk 1 above)
Comment (jiaoqingbo) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:57]: I need to treat the contents contained in the commandSet as a whole string, while separating the contents with spaces. this string as an argument to bash -C

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala (hunk 2 above)
Comment (jiaoqingbo) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:66]: I found that hive is passed through the -D parameter, so it remains consistent with hive. Do I also need to change hive's?

Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:66]: yes

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala (hunk 1 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:57]: i don't quite get it

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala (hunk 2 above)
Comment (jiaoqingbo) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:66]: OK I will change it later

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**